src/js/events.js

```diff
@@ -42,8 +42,13 @@ EventEmitter.prototype.emit = function(type) {
 
   var listeners = this._events[type];
   if (util.isArray(listeners)) {
-    listeners = listeners.slice();
```

Comment:
I dunno why this slice does exist, @legendecas @lolBig do you guess the reason?

Reply:
Hmmm, the CI failure seems to be caused by this deleted line.

Reply:
seems this slice is unnecessary

Reply:
on the other hand, the value of the listeners is only `Array` or `undefined`, can we use `if (listeners)` instead of `if (util.isArray(listeners))`,